src/utils/utils.js

```diff
@@ -679,6 +685,77 @@ const getStringId = (id) => {
     : JSON.stringify(id);
 };
 
+/**
+ * A function to validate and return Residency server input
+ * @returns string/undefined
+ */
+const getResidencyServer = (options) => {
+  const region = options ? options.residencyServer : undefined;
+  if (region) {
+    if (typeof region !== 'string' || !RESIDENCY_SERVERS.includes(region.toUpperCase())) {
+      logger.error('Invalid residencyServer input');
+      return undefined;
+    }
+    return region.toUpperCase();
+  }
+  return undefined;
+};
+
+const isValidServerUrl = (serverUrl) => {
+  if (!serverUrl || typeof serverUrl !== 'string' || serverUrl.trim().length === 0) {
+    return false;
+  }
+  return true;
+};
+
+/**
+ * A function to get url from source config response
+ * @param {array} urls    An array of objects containing urls
+ * @returns
+ */
+const getDefaultUrlofRegion = (urls) => {
+  let url;
+  if (Array.isArray(urls) && urls.length) {
+    const obj = urls.find((elem) => elem.default === true);
+    if (obj && isValidServerUrl(obj.url)) {
+      return obj.url;
+    }
+  }
+  return url;
+};
+
+/**
+ * A function to determine the dataPlaneUrl
+ * @param {Object} dataPlaneUrls An object containing dataPlaneUrl for different region
+ * @returns string
+ */
+const resolveDataPlaneUrl = (response, serverUrl, options) => {
+  try {
+    const dataPlanes = response.source.dataplanes || {};
+    // Check if dataPlanes object is present in source config
+    if (Object.keys(dataPlanes).length) {
+      const inputRegion = getResidencyServer(options);
+      const regionUrlArr = dataPlanes[inputRegion] || dataPlanes[DEFAULT_REGION];
+
+      if (regionUrlArr) {
+        const defaultUrl = getDefaultUrlofRegion(regionUrlArr);
+        if (defaultUrl) {
+          return defaultUrl;
+        }
+      }
+    }
+    // return the dataPlaneUrl provided in load API(if available)
+    if (isValidServerUrl(serverUrl)) {
+      return serverUrl;
+    }
+    // return the default dataPlaneUrl
+    return DEFAULT_DATAPLANE_URL;
```

Comment:
Lets change this to trow an error as it used to throw when data plane url was undefined or invalid.
Seems like product is yet to determine how this should behave so ets not introduce the auto fallback to hosted url yet.

Other sdks can do it since they do not autoupdate so breaking changes are esier to apply there.

will create a separate task only for these case

Reply:
throw Error('Unable to load the SDK due to invalid data plane url');

Reply:
Addressed